core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/registry/CachingCodecRegistry.java

```diff
@@ -302,9 +288,27 @@ protected CachingCodecRegistry(String logPrefix, TypeCodec<?>... userCodecs) {
     throw new CodecNotFoundException(null, javaType);
   }
 
+  // This is only used for the recursion from createCovariantCodec(GenericType)
+  protected TypeCodec<?> covariantCodecFor(GenericType<?> javaType) {
+    LOG.trace("[{}] Looking up codec for Java type {}", logPrefix, javaType);
+    for (TypeCodec<?> primitiveCodec : primitiveCodecs) {
+      if (primitiveCodec.getJavaType().isSupertypeOf(javaType)) {
+        LOG.trace("[{}] Found matching primitive codec {}", logPrefix, primitiveCodec);
+        return uncheckedCast(primitiveCodec);
+      }
+    }
+    for (TypeCodec<?> userCodec : userCodecs) {
+      if (userCodec.getJavaType().isSupertypeOf(javaType)) {
+        LOG.trace("[{}] Found matching user codec {}", logPrefix, userCodec);
+        return uncheckedCast(userCodec);
+      }
+    }
+    return uncheckedCast(getCachedCodec(null, javaType));
+  }
+
```

Comment:
Please don't move stuff around for no reason, this is really hard to review. There are no structural changes to this method, yet I had to do a line-by-line comparison.

Reply:
Hmm don't you think it's easy to follow now that this method is declared below the method where it is invoked from?

Reply:
Not necessarily, one might prefer to keep all the "codecFor" methods together because they do the same thing (this was probably my train of thought at the time).

But that's not the point, it's about minimising changes to streamline reviews.

Reply:
Fair enough, my apologies. And besides, I need this method exposed publicly so my argument in favor of moving it doesn't hold anymore.